src/base/lib/Convex/BuildTx.hs

```diff
@@ -108,71 +113,78 @@ import           Convex.MonadLog            (MonadLog (..), MonadLogIgnoreT,
 import           Convex.Scripts             (toScriptData)
 import           Data.Functor.Identity      (Identity (..))
 import           Data.List                  (nub)
-import           Data.Map                   (Map)
 import qualified Data.Map                   as Map
-import           Data.Maybe                 (fromMaybe)
-import           Data.Set                   (Set)
+import           Data.Maybe                 (fromMaybe, fromJust)
 import qualified Data.Set                   as Set
 import qualified Plutus.V1.Ledger.Api       as Plutus
 
-{-| A map of all inputs of the final transaction produced by a @TxBuilder@
--}
--- TOOO: This is essentially a subset of @TxBodyContent BuildTx@. Maybe we should
---       just make the entire @TxBodyContent BuildTx@ available to 'addInput'?
-data TransactionInputs = TransactionInputs
-  { txiSpendingInputs  :: Map C.TxIn (Witness WitCtxTxIn C.BabbageEra) -- ^ Inputs spent by the final transaction
-  , txiReferenceInputs :: Set C.TxIn -- ^ Reference inputs used by the final transaction
-  }
-
-mkTxInputs :: C.TxBodyContent C.BuildTx C.BabbageEra -> TransactionInputs
-mkTxInputs TxBodyContent{txIns, txInsReference} =
-  TransactionInputs
-    { txiSpendingInputs  = Map.fromList (fmap (view L._BuildTxWith) <$> txIns)
-    , txiReferenceInputs = Set.fromList (view L._TxInsReference txInsReference)
-    }
+type TxBody = C.TxBodyContent C.BuildTx C.BabbageEra
```

Comment:
Happy to change it if you feel strongly about this, but I generally think that it might occasionally be a good tradeoff. In this case there would be a bunch of `C.TxBodyContent C.BuildTx C.BabbageEra` all over the place making the types less readable imo.

Reply:
I'm for keeping this too, IMO the threshold for newtypes should be if we have several different synonyms for the same type, or if we want to distinguish values that have been validated in some way.

> type aliases don't expand consistently

Is this a tooling issue (hls) or GHC itself?